lib/AST/Pattern.cpp

```diff
@@ -400,6 +400,30 @@ SourceRange TuplePattern::getSourceRange() const {
            Fields.back().getPattern()->getEndLoc() };
 }
 
+TypedPattern::TypedPattern(Pattern *pattern, TypeRepr *tr,
+                           Optional<bool> implicit)
+  : Pattern(PatternKind::Typed), SubPattern(pattern), PatTypeRepr(tr) {
+  if (implicit.getValueOr(false) || !tr || !tr->getSourceRange().isValid())
```

Comment:
This is a different test from what was happening before; this treats everything that's deserialized as implicit. I don't think we want that.

Reply:
Good catch!